### ISSUE TITLE
Tests fail with the default GC in drchaos/common.nim:304:243 and mutator.nim:464:28

### ISSUE THREAD

**Issue by planetis-m** [status-im/nim-drchaos#8]
Both are caused by nil pointers. For the first one I made a bug report in https://github.com/nim-lang/Nim/issues/20278

**Comment by planetis-m** [comment]
Case object transitions are actually the culprit for the second. The only way to fix this is:

```nim
  var kindTmp = tmpSym.kindSym
  mFunc(kindTmp)
  when defined(gcDestructors):
    {.cast(uncheckedAssign).}:
      tmpSym.kindSym = kindTmp
  else:
    tmpSym = (typeof tmpSym)(kindSym: kindTmp)
```

The test run but the default gc may report misaligned address errors, i.e:
```
Nim/lib/system/gc_common.nim:394:33: runtime error: load of misaligned address 0x7ffd3bd64404 for type 'void *', which requires 8 byte alignment
```

It's also incorrect (previous fields are ignored). `-d:nimOldCaseObjects` caused even more errors. The only thing that works is --mm:arc/orc.

**Comment by planetis-m** [comment]
Documented in the readme.

**Comment by ba0f3** [comment]
I also getting this error with Nim 1.6.6: `/nim-drchaos/drchaos/common.nim(88, 32) Error: expression has no address; maybe use 'unsafeAddr'`

**Comment by planetis-m** [comment]
Thanks for reporting this is fixed in v0.1.4
